test/linter/test-notes.js

```diff
@@ -0,0 +1,217 @@
+'use strict';
+const chalk = require('chalk');
+const parser = require('node-html-parser');
+const { HtmlValidate } = require('html-validate');
+const { VALID_ELEMENTS } = require('../utils.js');
+
+const validator = new HtmlValidate();
+
+/**
+ * @typedef {import('../../types').Identifier} Identifier
+ *
+ * @typedef {'disallowed' | 'attrs' | 'attrs_a' | 'invalid' | 'doublespace'} HTMLErrorType
+ *
+ * @typedef {object} HTMLError
+ * @property {HTMLErrorType} errortype The type of error
+ * @property {string} relPath The identifier of the feature
+ * @property {string} browser The browser where the error was found
+ * @property {string} tag The specific HTML tag
+ * @property {string[]} messages Messages to describe the error
+ */
+
+/**
+ * Test a specific HTML tag (node) in notes for issues
+ *
+ * @param {object} node The HTML node to test
+ * @param {string} browser The browser the notes belong to
+ * @param {string} relPath The identifier of the feature
+ * @param {HTMLError[]} errors The array to push errors to
+ * @returns {void}
+ */
+const testNode = (node, browser, relPath, errors) => {
+  if (node.nodeType == 1) {
+    const tag = node.tagName?.toLowerCase();
+    if (tag && !VALID_ELEMENTS.includes(tag))
+      errors.push({
+        type: 'disallowed',
+        relPath,
+        browser,
+        tag,
+      });
+    if (tag !== 'a' && Object.entries(node.attributes).length !== 0) {
+      errors.push({
+        type: 'attrs',
+        relPath,
+        browser,
+        tag,
+      });
+    }
+    if (
+      tag === 'a' &&
+      (Object.entries(node.attributes).length !== 1 ||
+        Object.entries(node.attributes)[0][0] !== 'href')
+    ) {
+      errors.push({
+        type: 'attrs_a',
+        relPath,
+        browser,
+        tag,
+      });
+    }
+  }
+  for (let childNode of node.childNodes) {
+    testNode(childNode, browser, relPath, errors);
+  }
+};
+
+/**
+ * Test a string for valid HTML
+ *
+ * @param {string} string The string to test
+ * @param {string} browser The browser the notes belong to
+ * @param {string} relPath The identifier of the feature
+ * @param {HTMLError[]} errors The array to push errors to
+ * @returns {void}
+ */
+const validateHTML = (string, browser, relPath, errors) => {
+  const report = validator.validateString(string, {
+    rules: {
+      'attr-quotes': 'off',
+    },
+  });
+  if (report.valid) {
+    // If HTML is valid, ensure we're only using valid elements
```

Comment:
Can we unconditionally do this validation step? There's no input string that doesn't parse into *something*, valid or not.